include/kllvm/codegen/Decision.h

```diff
@@ -69,60 +82,39 @@ class SwitchNode : public DecisionNode {
   const std::vector<DecisionCase> &getCases() const { return cases; }
   
   virtual void codegen(Decision *d, llvm::StringMap<llvm::Value *> substitution);
-};
-
-class MakeLiteralNode : public DecisionNode {
-private:
-  std::string name;
-  ValueType cat;
-  std::string literal;
-  DecisionNode *child;
-
-  MakeLiteralNode(
-    const std::string &name,
-    ValueType cat,
-    const std::string literal,
-    DecisionNode *child) :
-      name(name),
-      cat(cat),
-      literal(literal),
-      child(child) {}
-
-public:
-  static MakeLiteralNode *Create(
-      const std::string &name,
-      ValueType cat,
-      const std::string literal,
-      DecisionNode *child) {
-    return new MakeLiteralNode(name, cat, literal, child);
-  }
-
-  virtual void codegen(Decision *d, llvm::StringMap<llvm::Value *> substitution);
+  virtual void collectUses(std::set<std::string> &vars) { if(cases.size() != 1 || cases[0].getConstructor()) vars.insert(name); for (auto _case : cases) { _case.getChild()->collectUses(vars); } }
```

Comment:
Why do either of these conditions guard whether `name` should be added to vars?

Reply:
if cases.size() == 0 && !cases[0].getConstructor(), then we don't actually generate a switch statement, just an unconditional branch to another block, so the variable isn't actually used by the generated code.

Reply:
Ok, I see that `case x of y -> ...` can be handled just by setting up the substitution without any instructions that actually read.

But if `cases.size()` is 0, isn't `cases[0]` reading past the end of the list?